### DISCUSSION TITLE
Use Projection with Features

### DISCUSSION THREAD

**Discussion by Kylar182** [NetTopologySuite/NetTopologySuite#755] [Q&A] [answered]
Let me preface this by saying that I'm new to Geography / Geometry. 

So, I'm trying to convert Esri Shape files into GeoJSON format. I can pull the DBF / SHP files with your library and convert to JSON but the input / output isn't in a format I can use. I can pull the PRJ file with another library (ProjNet) but I don't know how to use it. The shape files include dbf, prj, sbn, sbx, shx, and shp.

Using an online converter I was able to get the correct format that worked for my map component. It looks something like this

```json
{"type":"FeatureCollection", "features":[
{"type":"Feature","geometry":{"type":"Polygon","coordinates":[[[-85.15569710526616,37.491523204434515,0],...]]},
"properties":{"CNTY":"078","NAME":"BRADFORDSVILLE","COUNTY":"MARION","CRASH_CODE":"07806"}},
{"type":"Feature","geometry":{"type":"MultiPolygon","coordinates":[[[[-83.68876257422907,37.480259489414856,0],...]]]},"properties":{"CNTY":"095","NAME":"BOONEVILLE","COUNTY":"OWSLEY","CRASH_CODE":"09501"}},
]}
```
and it worked. 

The issue is that it only works for small files and I'd have to do it manually every time instead of dynamically in that app.

**Comment by FObermaier** [comment, marked as answer]
Using ProjNet you need to set up a CoordinateTransformation from your input-CoordinateSystem to the WGS84-CoordinateSystem.
The definition of your input-CoordinateSystem is written in the PRJ-file you mentioned. ProjNet has the capability of parsing it:

```C#
// Coordinate systems
var csInput = new ProjNet.CoordinateSystems.CoordinateSystemFactory().CreateFromWkt(prjText);
var csWGS48 = ProjNet.CoordinateSystems.GeographicCoordinateSystem.WGS84;

// Coordinate transformation
var ct = new ProjNet.CoordinateSystems.Transformations.CoordinateTransformationFactory().CreateFromCoordinateSystems(csInput, csWGS84);

// Apply transformation to feature's geometry
var mtf = new MathTransformFilter(ct.MathTransform);
foreach(var f in featuresFromSomewhere) {
   var geom = f.Geometry.Copy();
   geom.Apply(mtf);
   f.Geometry = geom;
}

//////////////////////////////////////////////////////////////////////////////////////
private class MathTransformFilter : ICoordinateSequenceFilter
{
    private readonly MathTransform _transform;

    public MathTransformFilter(MathTransform transform)
        => _transform = transform;

    public bool Done => false;
    public bool GeometryChanged => true;

    public void Filter(CoordinateSequence seq, int i)
    {
        var x = seq.GetX(i);
        var y = seq.GetY(i);
        var z = seq.GetZ(i);
        _transform.Transform(ref x, ref y, ref z);
        seq.SetX(i, x);
        seq.SetY(i, y);
        seq.SetZ(i, z);
    }
}
```

See also: https://github.com/NetTopologySuite/NetTopologySuite/wiki/The-visitor-pattern-in-NetTopologySuite#in-place-coordinate-transformation-with-ientirecoordinatesequencefilter

**Reply by Kylar182** [reply]
This was very helpful, I've now transcribed all my old shape files to GeoJSON!
